aws/resource_aws_gamelift_fleet.go

```diff
@@ -0,0 +1,596 @@
+	if len(cfg) < 1 {
+		return nil
+	}
+	out := gamelift.ResourceCreationLimitPolicy{}
+	m := cfg[0].(map[string]interface{})
+
+	if v, ok := m["new_game_sessions_per_creator"]; ok {
+		out.NewGameSessionsPerCreator = aws.Int64(int64(v.(int)))
+	}
+	if v, ok := m["policy_period_in_minutes"]; ok {
+		out.PolicyPeriodInMinutes = aws.Int64(int64(v.(int)))
+	}
+
+	return &out
+}
+
+func flattenGameliftResourceCreationLimitPolicy(policy *gamelift.ResourceCreationLimitPolicy) []interface{} {
+	if policy == nil {
+		return []interface{}{}
+	}
+
+	m := make(map[string]interface{}, 0)
+	m["new_game_sessions_per_creator"] = *policy.NewGameSessionsPerCreator
+	m["policy_period_in_minutes"] = *policy.PolicyPeriodInMinutes
+
+	return []interface{}{m}
+}
+
+func expandGameliftRuntimeConfiguration(cfg []interface{}) *gamelift.RuntimeConfiguration {
+	if len(cfg) < 1 {
+		return nil
+	}
+	out := gamelift.RuntimeConfiguration{}
+	m := cfg[0].(map[string]interface{})
+
+	if v, ok := m["game_session_activation_timeout_seconds"].(int); ok && v > 0 {
+		out.GameSessionActivationTimeoutSeconds = aws.Int64(int64(v))
+	}
+	if v, ok := m["max_concurrent_game_session_activations"].(int); ok && v > 0 {
+		out.MaxConcurrentGameSessionActivations = aws.Int64(int64(v))
+	}
+	if v, ok := m["server_process"]; ok {
+		out.ServerProcesses = expandGameliftServerProcesses(v.([]interface{}))
+	}
+
+	return &out
+}
+
+func expandGameliftServerProcesses(cfgs []interface{}) []*gamelift.ServerProcess {
+	if len(cfgs) < 1 {
+		return []*gamelift.ServerProcess{}
+	}
+
+	processes := make([]*gamelift.ServerProcess, len(cfgs), len(cfgs))
+	for i, rawCfg := range cfgs {
+		cfg := rawCfg.(map[string]interface{})
+		process := &gamelift.ServerProcess{
+			ConcurrentExecutions: aws.Int64(int64(cfg["concurrent_executions"].(int))),
+			LaunchPath:           aws.String(cfg["launch_path"].(string)),
+		}
+		if v, ok := cfg["parameters"].(string); ok && len(v) > 0 {
+			process.Parameters = aws.String(v)
+		}
+		processes[i] = process
+	}
+	return processes
+}
+
+func getGameliftFleetFailures(conn *gamelift.GameLift, id string) ([]*gamelift.Event, error) {
+	var events []*gamelift.Event
+	err := _getGameliftFleetFailures(conn, id, nil, &events)
+	return events, err
+}
+
+func _getGameliftFleetFailures(conn *gamelift.GameLift, id string, nextToken *string, events *[]*gamelift.Event) error {
+	eOut, err := conn.DescribeFleetEvents(&gamelift.DescribeFleetEventsInput{
+		FleetId:   aws.String(id),
+		NextToken: nextToken,
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, e := range eOut.Events {
+		if isGameliftEventFailure(e) {
+			*events = append(*events, e)
+		}
+	}
+
+	if eOut.NextToken != nil {
+		err := _getGameliftFleetFailures(conn, id, nextToken, events)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isGameliftEventFailure(event *gamelift.Event) bool {
+	failureCodes := []string{
+		"FLEET_STATE_ERROR",
```

Comment:
Attack of the SDK constants! 👽  (it'd be nice to lexicographically sort them too)

```
gamelift.EventCodeFleetStateError,
gamelift.EventCodeFleetInitializationFailed,
gamelift.EventCodeFleetBinaryDownloadFailed,
gamelift.EventCodeFleetValidationLaunchPathNotFound,
gamelift.EventCodeFleetValidationExecutableRuntimeFailure,
gamelift.EventCodeFleetValidationTimedOut,
gamelift.EventCodeFleetActivationFailed,
gamelift.EventCodeFleetActivationFailedNoInstances,
gamelift.EventCodeServerProcessInvalidPath,
gamelift.EventCodeServerProcessSdkInitializationTimeout,
gamelift.EventCodeServerProcessProcessReadyTimeout,
gamelift.EventCodeServerProcessCrashed,
gamelift.EventCodeServerProcessTerminatedUnhealthy,
gamelift.EventCodeServerProcessForceTerminated,
gamelift.EventCodeServerProcessProcessExitTimeout,
gamelift.EventCodeGameSessionActivationTimeout,
gamelift.EventCodeFleetVpcPeeringFailed,
```